### PR TITLE
Create a new Python Package pipeline for CUDA 12

### PR DESCRIPTION
**Pull request by jchen351**

### CODE REVIEW

**Comment by tianleiwu** [comment]
Great. Could we publish some nightly or weekly python package for CUDA 12?

**Comment by jchen351** [comment]
> Could we publish nightly or weekly python package for CUDA 12.1 or 12.2?

That is TBD, I am currently working publish it for Official release.

**Comment by snnn** [comment]
Very surprising that this new pipeline's Linux GPU build task is 3x faster than the existing one in https://aiinfra.visualstudio.com/Lotus/_build?definitionId=841&_a=summary .  Please carefully check if everything still seems good.

File: tools/ci_build/github/linux/build_linux_python_package.sh [hunk 1]
```diff
@@ -10,14 +10,17 @@ EXTRA_ARG=""
...
 do case "${parameter_Option}"
 in
 #GPU or CPU.
 d) BUILD_DEVICE=${OPTARG};;
-p) PYTHON_EXES=(${OPTARG});;
-x) EXTRA_ARG=(${OPTARG});;
+p) PYTHON_EXES=${OPTARG};;
+x) EXTRA_ARG=${OPTARG};;
 c) BUILD_CONFIG=${OPTARG};;
+u) CUDA_VERSION=${OPTARG};;
```
Comment (snnn) [inline tools/ci_build/github/linux/build_linux_python_package.sh:21]: You do not need to add an extra parameter here. There is an env var with the same name: “CUDA_VERSION” . You may just use that one.

File: tools/ci_build/github/linux/run_python_dockerbuild.sh [hunk 2]
```diff
@@ -2,31 +2,44 @@
...
 
-while getopts "i:d:x:c:" parameter_Option
+while getopts "i:d:x:c:u:" parameter_Option
 do case "${parameter_Option}"
 in
 i) DOCKER_IMAGE=${OPTARG};;
 d) DEVICE=${OPTARG};;
 x) BUILD_EXTR_PAR=${OPTARG};;
 c) BUILD_CONFIG=${OPTARG};;
+u) CUDA_VERSION=${OPTARG};;
```
Comment (snnn) [inline tools/ci_build/github/linux/run_python_dockerbuild.sh:12]: You do not need to add this flag

Comment (snnn) [reply tools/ci_build/github/linux/run_python_dockerbuild.sh:12]:
```bash
SHORT_CUDA_VERSION=$(echo $CUDA_VERSION | sed   's/\([[:digit:]]\+\.[[:digit:]]\+\)\.[[:digit:]]\+/\1/')
```
Then later on you can use it as
```
--cuda_version=$SHORT_CUDA_VERSION
```

File: tools/ci_build/github/linux/build_linux_python_package.sh (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/linux/build_linux_python_package.sh:21]:
```bash
SHORT_CUDA_VERSION=$(echo $CUDA_VERSION | sed   's/\([[:digit:]]\+\.[[:digit:]]\+\)\.[[:digit:]]\+/\1/')
```
Then later on you can use it as
```
--cuda_version=$SHORT_CUDA_VERSION
```

File: tools/ci_build/github/azure-pipelines/templates/py-linux-gpu.yml [hunk 3]
```diff
@@ -40,21 +63,22 @@ jobs:
...
 
 
     - task: Bash@3
       displayName: 'Build Python Wheel'
       inputs:
         targetType: filePath
         filePath: tools/ci_build/github/linux/run_python_dockerbuild.sh
-        # please check ONNXRUNTIME_CUDA_VERSION in tools/ci_build/github/linux/build_linux_arm64_python_package.sh
-        arguments: -i onnxruntimecuda118xtrt86build${{ parameters.arch }} -d "GPU" -c ${{ parameters.cmake_build_type }} -x "${{ parameters.extra_build_arg }}"
+        # please check ONNXRUNTIME_CUDA_VERSION in tools/ci_build/github/linux/build_linux_python_package.sh
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/py-linux-gpu.yml:79]: Remove this comment?

**Review by snnn — APPROVED** [review]

**Pull request merged**